### PR TITLE
Feat/admin search groups

### PR DESCRIPTION
**Pull request by duncanc19**
#### Work done
- Created search for group page in new admin section
- Created workplace-dropdown component used in search for group and to be used by search for workplace

#### Tests
Does this PR include tests for the changes introduced?
- [X] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts [hunk 1]
```diff
@@ -1,24 +1,265 @@
...
+
+      it('should collapse the Workplace details when clicking Close', async () => {
+        const { getByTestId } = await setup(true);
+
+        const searchResults = within(getByTestId('group-search-results'));
+
+        fireEvent.click(searchResults.getByText('Open'));
+        fireEvent.click(searchResults.getByText('Close'));
+
+        expect(searchResults.queryByTestId('groups-workplace-details')).toBeNull();
```
Comment (joannafawl) [inline src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts:201]: Would it be useful to check it's not null when you first click open?

File: src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,265 @@
...
+        const { fixture, searchGroupsSpy, getByTestId } = await setup();
+
+        searchGroupsSpy.and.returnValue(of([]));
+
+        const searchButton = getByTestId('searchButton');
+        fireEvent.click(searchButton);
+
+        fixture.detectChanges();
+
+        expect(getByTestId('no-search-results'));
```
Comment (joannafawl) [inline src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts:242]: Could you test the no search results text?

File: src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts (hunk 1 above)
Comment (duncanc19) [reply src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts:201]: The previous test checks that the content is there when it's open so I don't think we need it here too. I could change this to check for the text instead though to make the link between the tests clearer

File: src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts (hunk 2 above)
Comment (duncanc19) [reply src/app/features/admin/search/search-for-group/search-for-group.component.spec.ts]: Sure, I've just updated it

**Review by joannafawl — APPROVED** [review]

**Pull request merged**